### PR TITLE
ECR-1367 Import Guava tests

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview

Imported Guava tests with almost no changes.

---
See: https://jira.bf.local/browse/ECR-1367

### Definition of Done

- [x] There are no my TODOs left in the code
- [x] Change is covered by automated [tests](../CONTRIBUTING.md#tests)
- [x] The [coding guidelines](../CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] The [continuous integration build](https://www.travis-ci.org/exonum/exonum-java-binding) passes

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
Which version do these files correspond to?

File: exonum-java-proofs/src/test/java/com/exonum/binding/hash/HashingTest.java [hunk 1]
```diff
@@ -1,4 +1,19 @@
-/* 
+/*
```
Comment (dmitry-timofeev) [inline exonum-java-proofs/src/test/java/com/exonum/binding/hash/HashingTest.java]: Probably, ours is first.

**Comment by MakarovS** [comment]
> Which version do these files correspond to?

`25.1-jre` - the one we use.

File: exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+/**
+ * Skeleton implementation of {@link HashFunction}, appropriate for non-streaming algorithms. All
+ * the hash computation done using {@linkplain #newHasher()} are delegated to the {@linkplain
+ * #hashBytes(byte[], int, int)} method.
+ *
+ * @author Dimitris Andreou
+ */
+@Immutable
+abstract class AbstractNonStreamingHashFunction extends AbstractHashFunction {
```
Comment (dmitry-timofeev) [inline exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java]: Is this (and the second below) class required to test some of the functionality we already have?

Comment (MakarovS) [reply exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java]: Yes, we need that for `AbstractStreamingHasher`, which in turn we need for `FunnelsTest`.

**Comment by MakarovS** [comment]
> These classes use JUnit 3 APIs :)

Do you suggest we refactor those? I tried to keep any changes to minimum.

File: exonum-java-proofs/src/test/java/com/exonum/binding/hash/HashingTest.java (hunk 1 above)
Comment (MakarovS) [reply exonum-java-proofs/src/test/java/com/exonum/binding/hash/HashingTest.java]: In other guava classes we have Guava license first, ours second. Do you think we should change it there as well?

File: exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java (hunk 2 above)
Comment (dmitry-timofeev) [reply exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java]: I see, do you think it makes sense to put these files in `test/` folder, or keep here in `main/` to be consistent with the source of this package?

**Comment by dmitry-timofeev** [comment]
> Do you suggest we refactor those? I tried to keep any changes to minimum.

No, certainly not. We may consider doing that system-wide when will migrate to JUnit 5 in the future.

File: exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java (hunk 2 above)
Comment (MakarovS) [reply exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java]: I think we should keep those in main for consistency with original Guava lib. Plus someone might want to use those outside of tests.

Comment (dmitry-timofeev) [reply exonum-java-proofs/src/main/java/com/exonum/binding/hash/AbstractNonStreamingHashFunction.java]: OK, the latter is unlikely for they are package-private.

**Pull request merged**